evm/deploy-contract.js

```diff
@@ -170,92 +157,43 @@ async function deploy(options, chain) {
-            break;
-        }
-
-        case 'create3': {
-            printInfo(`${contractName} deployment salt`, salt);
+    let deployerContract = deployMethod === 'create3' ? contracts.Create3Deployer?.address : contracts.ConstAddressDeployer?.address;
 
-            create3Deployer = contracts.Create3Deployer?.address;
+    if (deployMethod === 'create') {
+        deployerContract = null;
+    }
 
-            if (!create3Deployer) {
-                throw new Error(`Create3 deployer does not exist on ${chain.name}.`);
-            }
+    const predictedAddress = await getDeployedAddress(wallet.address, deployMethod, {
+        salt,
+        deployerContract,
+        contractJson,
+        constructorArgs,
+        provider: wallet.provider,
+    });
 
-            const contractAddress = await getCreate3Address(create3Deployer, wallet.connect(provider), salt);
-            printInfo(`${contractName} will be deployed to`, contractAddress);
-            break;
-        }
+    if (deployMethod !== 'create') {
+        printInfo(`${contractName} deployment salt`, salt);
     }
 
+    printInfo('Deployment method', deployMethod);
+    printInfo('Deployer contract', deployerContract);
+    printInfo(`${contractName} will be deployed to`, predictedAddress);
+
     if (!yes) {
         console.log('Does this match any existing deployments?');
         const anwser = readlineSync.question(`Proceed with deployment on ${chain.name}? ${chalk.green('(y/n)')} `);
         if (anwser !== 'y') return;
     }
 
-    let contract;
-
-    switch (deployMethod) {
-        case 'create': {
-            contract = await deployContract(wallet, contractJson, constructorArgs, gasOptions, verifyOptions);
-            break;
-        }
-
-        case 'create2': {
-            contract = await deployCreate2(
-                constAddressDeployer,
-                wallet,
-                contractJson,
-                constructorArgs,
-                salt,
-                gasOptions.gasLimit,
-                verifyOptions,
-            );
-
-            contractConfig.salt = salt;
-            printInfo(`${chain.name} | ConstAddressDeployer`, constAddressDeployer);
-            break;
-        }
-
-        case 'create3': {
-            contract = await deployCreate3(
-                create3Deployer,
-                wallet.connect(provider),
-                contractJson,
-                constructorArgs,
-                salt,
-                gasOptions,
-                verifyOptions,
-            );
-
-            contractConfig.salt = salt;
-            printInfo(`${chain.name} | Create3Deployer`, create3Deployer);
-            break;
-        }
-    }
+    const contract = await deployContract(
+        deployMethod,
+        wallet,
+        contractJson,
+        constructorArgs,
+        { salt, deployerContract },
+        gasOptions,
+        verifyOptions,
+    );
 
```

Comment:
Do we still want to store the salt in the info file for create2/create3 methods?

Reply:
ah, good catch, forgot to refactor that